### PR TITLE
carousel: add missing `to` description and fix `nextwhenvisible` description

### LINKED ISSUE #31788: Confusion on the "nextWhenVisible()" method on Carousel Component
Hi everyone, I am currently using bootstrap 5 and while browsing through documentation, I have found that in the carousel component documentation, `carousel.prototype.nextWhenVisible()` function is not accurate with its description. Although `carousel.prototype.to()` function seems to be matched with the description (which is not found in the carousel component documentation, but can be found in the code).

I think that the function has been mismatched with the description, that have led me to a confusion.

The section that I am referring to:
https://v5.getbootstrap.com/docs/5.0/components/carousel/#methods

### PR DESCRIPTION
**Pull request by cankitm**
Added the to method and corrected nextWhenVisible description in Corousel Component documentation.

Fixes #31788

### CODE REVIEW

**Comment by XhmikosR** [comment]
BTW this is duplicate of #31800

**Comment by cankitm** [comment]
> BTW this is duplicate of #31800

#31800 was showing several unwanted changes, so opened a new PR, specifically related to the issue.

File: site/content/docs/5.0/components/carousel.md [hunk 1]
```diff
@@ -364,6 +364,10 @@ var carousel = new bootstrap.Carousel(myCarousel, {
     </tr>
     <tr>
       <td><code>nextWhenVisible</code></td>
+      <td>Don't cycle carousel to next if it's not available. <strong>Returns to the caller before the target item has been shown</strong>
```
Comment (Johann-S) [inline site/content/docs/5.0/components/carousel.md:367]: we should mention it use the visible API here but not the `next` method

Comment (cankitm) [reply site/content/docs/5.0/components/carousel.md]: Updated to -->
Don't cycle carousel to next when the page isn't visible or the carousel or its parent isn't visible. Returns to the caller before the target item has been shown

**Review by Johann-S — APPROVED** [review]

File: site/content/docs/5.0/components/carousel.md [hunk 2]
```diff
@@ -364,6 +364,10 @@ var carousel = new bootstrap.Carousel(myCarousel, {
     </tr>
     <tr>
       <td><code>nextWhenVisible</code></td>
+      <td>Don't cycle carousel to next when the page isn't visible or the carousel or its parent isn't visible. <strong>Returns to the caller before the target item has been shown</strong>
```
Comment (XhmikosR) [inline site/content/docs/5.0/components/carousel.md:367]: @mdo or @patrickhlauke for word proofing. I feel like a "the" is missing?

**Comment by XhmikosR** [comment]
@Johann-S does this need a backport or is the option present only in v5?

**Comment by Johann-S** [comment]
yep we should backport this in v4 too 👍

**Comment by XhmikosR** [comment]
I don't see the option at all in v4-dev docs, so I guess we'll need to manually backport it. I'll add it to the 4.5.3 project but any patch to backport it is welcome.

**Comment by cankitm** [comment]
> I don't see the option at all in v4-dev docs, so I guess we'll need to manually backport it. I'll add it to the 4.5.3 project but any patch to backport it is welcome.

Manual Backport : #31851

**Pull request merged**